--- /tmp/CMakeLists.txt	2020-05-07 03:36:26.811561077 +0000
+++ ./plugins/connection/CMakeLists.txt	2020-05-07 03:37:16.737048618 +0000
@@ -16,6 +16,6 @@
 REGISTER_PLUGIN(TARGET redirection
                 TYPE MARIADB_CLIENT_PLUGIN_CONNECTION
                 CONFIGURATIONS STATIC DYNAMIC OFF
-                DEFAULT DYNAMIC
+                DEFAULT STATIC
                 SOURCES ${CC_SOURCE_DIR}/plugins/connection/my_redirect.c
                 ${CC_SOURCE_DIR}/plugins/connection/redirection.c)
