--- ../mariadb-connector-c-private/plugins/connection/redirection.c	2020-05-28 03:10:44.587309188 +0000
+++ deps/mariadb-client-library/mariadb-connector-c-3.1.4-redirect/plugins/connection/redirection.c	2020-05-28 04:04:02.300300017 +0000
@@ -239,6 +239,12 @@
         int port_len = p3 - p2 - 2;
         int user_len = strlen(info_str) - (p3 + 5 - info_str);
 
+        char* end_of_user = strstr(p3, "\025");
+        if (end_of_user)
+        {
+            user_len -= strlen(end_of_user);
+        }
+
         if (host_len <= 0 || port_len <= 0 || user_len <= 0)
         {
             return NULL;
@@ -469,43 +475,4 @@
     }
 
     mutex_unlock(redirect_mutex);
-}
-
-void *ma_multi_malloc(myf myFlags, ...)
-{
-  va_list args;
-  char **ptr,*start,*res;
-  size_t tot_length,length;
-
-  va_start(args,myFlags);
-  tot_length=0;
-  while ((ptr=va_arg(args, char **)))
-  {
-    length=va_arg(args, size_t);
-    tot_length+=ALIGN_SIZE(length);
-  }
-  va_end(args);
-
-  if (!(start=(char *)malloc(tot_length)))
-    return 0;
-
-  va_start(args,myFlags);
-  res=start;
-  while ((ptr=va_arg(args, char **)))
-  {
-    *ptr=res;
-    length=va_arg(args,size_t);
-    res+=ALIGN_SIZE(length);
-  }
-  va_end(args);
-  return start;
-}
-
-char *ma_strmake(register char *dst, register const char *src, size_t length)
-{
-  while (length--)
-    if (! (*dst++ = *src++))
-      return dst-1;
-  *dst=0;
-  return dst;
 }
\ No newline at end of file
